solidity/test/integration/Slashing.test.ts

```diff
@@ -0,0 +1,529 @@
+        })
+
+        it("should slash wallet members", async () => {
+          const { redemptionTimeoutSlashingAmount: amountToSlash } =
+            await bridge.redemptionParameters()
+
+          expect(await staking.getSlashingQueueLength()).to.equal(
+            walletMembers.length
+          )
+
+          for (let i = 0; i < walletMembers.length; i++) {
+            const slashing = await staking.slashingQueue(i)
+
+            expect(slashing.amount).to.equal(
+              amountToSlash,
+              `unexpected slashing amount for ${i}`
+            )
+
+            expect(slashing.stakingProvider).to.equal(
+              walletMembers[i].stakingProvider,
+              `unexpected staking provider for ${i}`
+            )
+          }
+        })
+
+        it("should not close the wallet in the wallet registry", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(await walletRegistry.isWalletRegistered(ecdsaWalletID)).to.be
+            .true
+        })
+
+        // Since the wallet's balance was above 0, it switched to MovingFunds state.
+        it("should transition the wallet in the bridge to the MovingFunds state", async () => {
+          const storedWallet = await bridge.wallets(walletPubKeyHash160)
+
+          expect(storedWallet.state).to.be.equal(walletState.MovingFunds)
+        })
+
+        it("should consume around 3 150 000 gas for Bridge.notifyRedemptionTimeout transaction", async () => {
+          await assertGasUsed(notifyRedemptionTimeoutTx, 3_150_000, 100_000)
+        })
+      })
+    })
+  })
+
+  describe("notifyMovingFundsTimeout", async () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    describe("when wallet is created", async () => {
+      const deposit = SingleP2SHDeposit.deposits[0]
+
+      const walletPubKeyHash160 = deposit.reveal.walletPubKeyHash
+      const { walletPublicKey, walletID: ecdsaWalletID } = deposit.ecdsaWallet
+
+      let walletMembers: Operators
+
+      before(async () => {
+        const requestNewWalletTx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+
+        await produceRelayEntry(walletRegistry, randomBeacon)
+        ;({ walletMembers } = await performEcdsaDkg(
+          walletRegistry,
+          walletPublicKey,
+          requestNewWalletTx.blockNumber
+        ))
+
+        const { fundingTx, reveal } = SingleP2SHDeposit.deposits[0]
+        reveal.vault = tbtcVault.address
+
+        // We use a deposit funding bitcoin transaction with a very low amount,
+        // so we need to update the dust threshold to be below it.
+        await updateDepositDustThreshold(10000) // 0.0001 BTC)
+
+        // Reveal and sweep the deposit to set up a main UTXO for the deposit,
+        // so when operator inactivity is reported the deposit is transferred to
+        // the MovingFunds instead of the Closing state.
```

Comment:
```suggestion
        // Reveal and sweep the deposit to set up a main UTXO for the wallet,
        // so when operator inactivity is reported the wallet is transferred to
        // the MovingFunds instead of the Closing state.
```